Comments on wheel building script

matthew-brett opened the pull request:
I think you actively don't want the pandas checkout to go into the host operating system, because you want the checkout to be a temporary copy that you will throw away with the building image.

The `--no-deps` flag avoids copying dependencies into the wheelhouse, such as numpy.

matthew-brett committed:
Comments on wheel building script

I think you actively don't want the pandas checkout to go into the host operating system, because you want the checkout to be a temporary copy that you will throw away with the building image.

The `--no-deps` flag avoids copying dependencies into the wheelhouse, such as numpy.

TomAugspurger commented:
Hi Matthew, thanks for looking. `--no-deps` does help.

I've been tinkering with it a bit more in https://github.com/TomAugspurger/pandas-manylinux/pull/2 and was able to get Travis to build and run the tests in a container without timing out. Would you be open to those uploading to wheels.scipy.org at the end of the build similar to https://github.com/MacPython/numpy-wheels? And then we can upload those to PyPI for each release?

matthew-brett commented:
Sure - no problem - I'll add the encrypted key to this PR.

matthew-brett commented:
Which repo will this be building from - TomAugspurger/pandas-manylinux ?

TomAugspurger commented:
Ahh, I think I'll do it from `pydata/pandas-manylinux`. I'll create that quick.

TomAugspurger commented:
Whoops, I don't have permission to create a new repo under pydata. @jreback I'm guessing you do? I think it make sense to have this live there, right? Could you make a blank `pydata/pandas-manylinux` repo and give the pandas devs access to it?

jreback commented:
https://github.com/pydata/pandas-manylinux

should be good

matthew-brett committed:
Add encrypted key to pandas-manylinux repo

This should upload the wheels when build triggered on that repo (and
only that repo).

matthew-brett commented:
Sorry to be a bit [NIH](https://en.wikipedia.org/wiki/Not_invented_here) - but I was working on a scripting system for combined OSX / Manylinux wheels over at https://github.com/matthew-brett/multibuild - and it turned out to be pretty easy to update the current Pandas wheel-building repo to build 64- and 32-bit manylinux wheels : 
- https://travis-ci.org/MacPython/pandas-wheels/builds/142424402
- https://github.com/MacPython/pandas-wheels

There are some 32-bit test failures (see https://github.com/pydata/pandas/issues/13566), but the 64-bit wheels test OK, and are now up at http://wheels.scipy.org/ .  See also updated info at https://github.com/MacPython/pandas-wheels .

Would you consider uploading these?   Numpy / scipy / Cython already have Manylinux wheels, matplotlib and h5py wheels are imminent.  Automated upload instructions at https://github.com/MacPython/pandas-wheels#uploading-the-built-wheels-to-pypi .

TomAugspurger commented:
Saw your post yesterday on the manylinux ML, you beat me to it :) Thanks for solving this in a general way.

I think Jeff has the permissions to upload to PyPI.

matthew-brett commented:
@jreback - anything I can do to help here?  Should I move the discussion over the pandas mailing list or issues?

matthew-brett commented:
@jreback - should I email the Pandas list?  Or open a Pandas issue about this?

stefanv commented:
@matthew-brett @jreback We'd very much appreciate if Pandas wheels can appear on PyPi.  It's slowing down our test suites significantly to have to rebuild from source each time.

jreback commented:
@matthew-brett I just saw your comment.

what is the difference between m & mu? (need both)?

```
pandas-0.18.1-cp27-cp27m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl  10Mi    2016-07-05 12:36:32
pandas-0.18.1-cp27-cp27m-manylinux1_x86_64.whl  14Mi    2016-07-05 12:18:17
pandas-0.18.1-cp27-cp27mu-manylinux1_x86_64.whl 14Mi    2016-07-05 12:19:26
pandas-0.18.1-cp34-cp34m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl  9Mi 2016-07-05 12:36:57
pandas-0.18.1-cp34-cp34m-manylinux1_i686.whl    14Mi    2016-07-05 12:30:47
pandas-0.18.1-cp34-cp34m-manylinux1_x86_64.whl  15Mi    2016-07-05 12:18:24
pandas-0.18.1-cp35-cp35m-macosx_10_6_intel.macosx_10_9_intel.macosx_10_9_x86_64.macosx_10_10_intel.macosx_10_10_x86_64.whl  9Mi 2016-07-05 12:46:37
pandas-0.18.1-cp35-cp35m-manylinux1_i686.whl    14Mi    2016-07-05 12:30:30
pandas-0.18.1-cp35-cp35m-manylinux1_x86_64.whl  14Mi    
```

jreback commented:
ok they are all uploaded.

matthew-brett commented:
`m` is for Python with a narrow unicode build (16-bit unicode).  These are a bit uncommon, but old `pyenv` builds were narrow, and in fact travis-ci trusty containers use older pyenv builds.  `mu` is for the more standard 32-bit unicode build that most (?all) distributions ship. This distinction disappears with the new unicode representation in Python 3.3.

matthew-brett commented:
Great - thanks for uploading them.  Please let me know if there's anything I can do to help with the next release.  I think that it should be as simple as the instructions at https://github.com/macpython/pandas-wheels#triggering-a-build

stefanv commented:
Thank you very much!
​
